### PR TITLE
Extra fields in audit log when publishing from GitHub Actions.

### PR DESCRIPTION
**Pull request by isoos**
#6094

### CODE REVIEW

File: app/lib/audit/models.dart [hunk 1]
```diff
@@ -209,10 +209,14 @@ class AuditLogRecord extends db.ExpandoModel<String> {
...
   }) {
+    final uploaderFields = uploader.fields;
+    final formattedFields = uploaderFields == null
+        ? ''
+        : ' (${uploaderFields.entries.map((e) => '${e.key}: `${e.value}`').join(', ')})';
     final summary = [
       'Package `$package` version `$version`',
       if (publisherId != null) ' owned by publisher `$publisherId`',
-      ' was published by `${uploader.displayId}`.',
+      ' was published by `${uploader.displayId}`$formattedFields.',
```
Comment (jonasfj) [inline app/lib/audit/models.dart:219]: This is a user-facing message, and security it's probably relatively important that users understand this message.
Might I suggest something along the lines of:
```markdown
Package `retry` version `1.2.3` was published from Github Actions running on revision `a2a2a2aa2a`
from the `google/dart-neats` repository.
The authenticating Github Actions flow was triggered by a user identified as `jonasfj` on Github (at the time of publishing).
```

Comment (isoos) [reply app/lib/audit/models.dart]: Updated. PTAL.

File: app/lib/audit/models.dart [hunk 2]
```diff
@@ -209,11 +209,37 @@ class AuditLogRecord extends db.ExpandoModel<String> {
...
     String? publisherId,
   }) {
-    final summary = [
+    final summaryParts = <String>[
       'Package `$package` version `$version`',
       if (publisherId != null) ' owned by publisher `$publisherId`',
-      ' was published by `${uploader.displayId}`.',
-    ].join();
+    ];
+    Map<String, dynamic>? fields;
```
Comment (jonasfj) [inline app/lib/audit/models.dart:216]:
```suggestion
    var fields = <String, dynamic>{};
```

File: app/lib/audit/models.dart [hunk 3]
```diff
@@ -226,6 +252,7 @@ class AuditLogRecord extends db.ExpandoModel<String> {
         'version': version,
         if (uploader is AuthenticatedUser) 'email': uploader.user.email,
         if (publisherId != null) 'publisherId': publisherId,
+        ...?fields,
```
Comment (jonasfj) [inline app/lib/audit/models.dart:255]:
```suggestion
        ...fields,
```

**Review by jonasfj — APPROVED** [review]

**Pull request merged**